imap_processing/lo/l0/data_classes/science_direct_events.py

```diff
@@ -0,0 +1,213 @@
+    left. All other fields do not need to be bit shifted.
+
+    The raw values are computed for L1A and will be converted to
+    engineering units in L1B.
+
+    Attributes
+    ----------
+    SHCOARSE : int
+        Spacecraft time.
+    COUNT: int
+        Number of direct events.
+    DATA: str
+        Compressed TOF Direct Event time tagged data.
+    TIME: numpy.ndarray
+        time tag for the direct event
+    ENERGY: numpy.ndarray
+        energy of the direct event ENA.
+    MODE: numpy.ndarray
+        Indication of how the data is packed.
+    TOF0: numpy.ndarray
+        Time of Flight 0 value for direct event.
+    TOF1: numpy.ndarray
+        Time of Flight 1 value for direct event.
+    TOF2: numpy.ndarray
+        Time of Flight 2 value for direct event.
+    TOF3: numpy.ndarray
+        Time of Flight 3 value for direct event.
+    CKSM: numpy.ndarray
+        This is checksum defined relative to the TOFs
+        condition for golden triples. If golden triples are below
+        a certain threshold in checksum it's considered golden, otherwise,
+        it's considered a silver triple. This is important for the compression
+        for golden triples because it's used to recover TOF1 because
+        compression scheme to save space on golden triples doesn't send
+        down TOF1 so it's recovered on the ground using the checksum
+    POS: numpy.ndarray
+        Stop position for the direct event. There are 4 quadrants
+        on the at the stop position.
+
+    Methods
+    -------
+    __init__(packet, software_vesion, packet_file_name):
+        Uses the CCSDS packet, version of the software, and
+        the name of the packet file to parse and store information about
+        the Direct Event packet data.
+    """
+
+    software_version: str
+    packet_file_name: str
+    ccsds_header: CcsdsData
+    SHCOARSE: int
+    COUNT: int
+    DATA: str
+    TIME: np.ndarray
+    ENERGY: np.ndarray
+    MODE: np.ndarray
+    TOF0: np.ndarray
+    TOF1: np.ndarray
+    TOF2: np.ndarray
+    TOF3: np.ndarray
+    CKSM: np.ndarray
+    POS: np.ndarray
+
+    def __init__(self, packet, software_version: str, packet_file_name: str):
+        """Initialize Science Direct Events Data class."""
+        set_attributes(self, packet)
+        self.software_version = software_version
+        self.packet_file_name = packet_file_name
+        self.ccsds_header = CcsdsData(packet.header)
+        # TODO: Is there a better way to initialize these arrays?
+        self.TIME = np.array([])
+        self.ENERGY = np.array([])
+        self.MODE = np.array([])
+        self.TOF0 = np.array([])
+        self.TOF1 = np.array([])
+        self.TOF2 = np.array([])
+        self.TOF3 = np.array([])
+        self.CKSM = np.array([])
+        self.POS = np.array([])
+        self._decompress_data()
+
+    def _decompress_data(self):
+        """Decompress the Lo Science Direct Events data."""
+        data = BinaryString(self.DATA)
```

Comment:
A stupid question. Where is this `self.DATA` and `self.COUNT` initialized? I feel like I am missing something.

Reply:
Not a stupid question! Right now it's being initialized in the `set_attributes` function which gets passed `self` and the `packet`. I made that update in this PR, but I'm going to go back to having a parent class `LoBase` that has a method that works the same as `set_attributes`. So it will work the same, but instead of calling `set_attributes` it will call that as a parent method.